android/src/main/java/fr/greweb/rnwebgl/RNWebGLView.java

```diff
@@ -1,96 +1,242 @@
-    mGLViewMap.remove(ctxId);
-    reactContext.getNativeModule(RNWebGLTextureLoader.class).unloadWithCtxId(ctxId);
-    RNWebGLContextDestroy(ctxId);
-    super.onDetachedFromWindow();
+  @Override
+  public void onSurfaceTextureSizeChanged(SurfaceTexture surface, int width, int height) {
   }
 
-  public synchronized void runOnGLThread(Runnable r) {
-    mEventQueue.add(r);
+  @Override
+  public void onSurfaceTextureUpdated(SurfaceTexture surface) {
   }
 
-  public synchronized static void runOnGLThread(int ctxId, Runnable r) {
-    RNWebGLView glView = mGLViewMap.get(ctxId);
-    if (glView != null) {
-      glView.runOnGLThread(r);
+
+  // All actual GL calls are made on this thread
+
+  private class GLThread extends Thread {
+    private final SurfaceTexture mSurfaceTexture;
+
+    private static final int EGL_CONTEXT_CLIENT_VERSION = 0x3098;
+    private EGLDisplay mEGLDisplay;
+    private EGLSurface mEGLSurface;
+    private EGLContext mEGLContext;
+    private EGL10 mEGL;
+
+    GLThread(SurfaceTexture surfaceTexture) {
+      mSurfaceTexture = surfaceTexture;
+    }
+
+    @Override
+    public void run() {
+      initEGL();
+
+      while (true) {
+        try {
+          makeEGLContextCurrent();
+
+          // Flush any queued events
+          for (Runnable r : mEventQueue) {
+            r.run();
+          }
+          mEventQueue.clear();
+
+          // ctxId may be unset if we get here (on the GL thread) before EXGLContextCreate(...) is
+          // called on the JS thread (see above in the implementation of `onSurfaceTextureAvailable(...)`)
+          if (ctxId > 0) {
+            RNWebGLContextFlush(ctxId);
+          }
+
+          if (!mEGL.eglSwapBuffers(mEGLDisplay, mEGLSurface)) {
+            Log.e("RNWebGLView", "cannot swap buffers!");
+          }
+          checkEGLError();
+
+          Thread.sleep(1000 / 60);
+        } catch (InterruptedException e) {
+          break;
+        }
+      }
+
+      deinitEGL();
+    }
+
+    private void initEGL() {
+      mEGL = (EGL10) EGLContext.getEGL();
+
+      // Get EGLDisplay and initialize display connection
+      mEGLDisplay = mEGL.eglGetDisplay(EGL10.EGL_DEFAULT_DISPLAY);
+      if (mEGLDisplay == EGL10.EGL_NO_DISPLAY) {
+        throw new RuntimeException("eglGetDisplay failed " + GLUtils.getEGLErrorString(mEGL.eglGetError()));
+      }
+      int[] version = new int[2];
+      if (!mEGL.eglInitialize(mEGLDisplay, version)) {
+        throw new RuntimeException("eglInitialize failed " + GLUtils.getEGLErrorString(mEGL.eglGetError()));
+      }
+
+      // Find a compatible EGLConfig
+      int[] configsCount = new int[1];
+      EGLConfig[] configs = new EGLConfig[1];
+      int[] configSpec = {
+              EGL10.EGL_RENDERABLE_TYPE, EGL14.EGL_OPENGL_ES2_BIT,
+              EGL10.EGL_RED_SIZE, 8, EGL10.EGL_GREEN_SIZE, 8, EGL10.EGL_BLUE_SIZE, 8,
+              EGL10.EGL_ALPHA_SIZE, 8, EGL10.EGL_DEPTH_SIZE, 0, EGL10.EGL_STENCIL_SIZE, 0,
```

Comment:
You're allocating the zbuffer with 0 bits here. I changed it to 16 bits in my code but I'm not sure what the value should be.

Reply:
I'm not sure either. Honestly the code is just taken from Expo's version, that this repo is forked off of. The code worked for me.